### ISSUE TITLE
Restored custom editors never load if the extension is uninstalled

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#125582]
Repro:

1. Install Luna Paint
2. Open an image
3. Uninstall Luna Paint
4. Reload the window

The editor will stay in loading animation forever:

<img width="579" alt="Screen Shot 2021-06-05 at 3 20 07 AM" src="https://user-images.githubusercontent.com/2193314/120888434-2246a000-c5ad-11eb-9758-7a3f4eb9efa4.png">

**Comment by lramos15** [comment]
@bpasero I'm not sure of the best path forward here so I'm adding you to the issue as well. This has to do with restoring editors in general. We call `this.restoreEdtiors` from `editorGroupView.ts` and try and show the custom editor or notebook editor input without the backing extension. This leads to the bug above. We cannot access the editorOverrideService from the group without creating a circular dependency.

**Comment by bpasero** [comment]
I do not agree that this is a problem the editor override service should solve because we are not opening an editor, we are restoring an editor and imho there it does not make sense to use a different editor for opening. The user picked Luna paint to open a file and I would expect to restore Luna paint as editor, not some other editor. 

I think this is something the custom editor should solve:
* either throw an error when the extension is not enabled (this will gracefully close the editor)
* or show a generic message in the editor that the extension either needs to be installed or enabled

**Comment by Tyriar** [comment]
There are two cases I think we should handle:

- When the restored editor is non-dirty: Reopen automatically using the new default editor since data loss is not a concern and it's what the user would probably expect after uninstalling the extension
- When the restored editor is dirty:
  > show a generic message in the editor that the extension either needs to be installed or enabled

**Comment by bpasero** [comment]
What if the extension is just temporarily disabled and later will be enabled again? You would essential loose that UI state?

**Comment by lramos15** [comment]
This is what notebooks provide and I think it's more than enough. They show a blank editor with nothing in it and just let the user know that this was extension backed.
<img width="474" alt="Screen Shot 2021-06-08 at 10 08 29 AM" src="https://user-images.githubusercontent.com/4544166/121200199-8c777300-c841-11eb-8906-7ab2422f5931.png">

**Comment by Tyriar** [comment]
Sounds good to me

**Comment by bpasero** [comment]
I would probably prefer a solution that is more inline without a notification (similar to our binary editor that suggests to reopen with another editor using a link in the content area.

**Comment by joyceerhl** [comment]
We now have this nice text whenever an editor cannot restore, I wonder if we could show a message like that for custom editors which fail to restore, with a link to the `Reopen With...` action.

<img width="770" alt="image" src="https://user-images.githubusercontent.com/30305945/190492532-70f1833f-153c-444d-9382-bc8b162d091f.png">

**Comment by sripavantejb** [comment]
Triaging this issue — it's been open since 2021 and is still relevant.

**Summary:** When a custom editor (e.g., Luna Paint for images) is restored on VS Code reload, but the backing extension has been uninstalled or disabled, the editor gets stuck in an infinite loading spinner with no way to recover or close.

**Current state based on the thread:**
- The team reached consensus on the preferred UX: show an inline message in the editor area (similar to the binary editor's "Reopen With..." approach) rather than a notification
- - This approach gracefully handles both dirty and non-dirty editors
- - - Related issue #161008 was mentioned as tracking a similar notebook case
**Suggested implementation direction (for contributors):**
When `restoreEditors` is called and a custom editor's contributing extension is not available, instead of showing an infinite spinner, the workbench could render a fallback UI in the editor content area with:
1. A message like: "This editor requires the '[Extension Name]' extension, which is not installed or is disabled."
2. 2. A "Reopen With Default Editor" action link for non-dirty files
3. 3. For dirty files: a recovery option or instructions to re-enable the extension
The `joyceerhl` comment already pointed to the pattern that now exists for generic editor restore failures — it would make sense to hook into that same mechanism for the custom editor case.

Happy to help investigate or test if this is still unaddressed.
